tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4804 @@
+        } else if (OB_ISNULL(backup_set_attr->backup_set_store_)) {
+          ret = OB_ERR_UNEXPECTED;
+          STORAGE_LOG(WARN, "backup set store is null", K(ret), K(backup_set_id));
+        } else if (OB_ISNULL(tablet_attr->macro_block_id_mappings_meta_index_)) {
+          ret = OB_ERR_UNEXPECTED;
+          STORAGE_LOG(WARN, "tablet meta index is null", K(ret));
+        } else if (OB_ISNULL(alc_ptr = local_allocator_.alloc(
+                                 sizeof(backup::ObBackupMacroBlockIDMappingsMeta)))) {
+          ret = OB_ALLOCATE_MEMORY_FAILED;
+          STORAGE_LOG(WARN, "failed to alloc backup::ObBackupMacroBlockIDMappingsMeta", K(ret));
+        } else if (FALSE_IT(id_mappings_meta_.at(i)
+                            = new (alc_ptr) backup::ObBackupMacroBlockIDMappingsMeta())) {
+        } else if (OB_FAIL(ObAdminBackupValidationUtil::read_macro_block_id_mappings_meta(
+                       backup_set_attr->backup_set_store_->get_backup_set_dest(),
+                       *tablet_attr->macro_block_id_mappings_meta_index_, tablet_attr->data_type_,
+                       *id_mappings_meta_.at(i), ctx_))) {
+          STORAGE_LOG(WARN, "failed to read id mappings meta", K(ret), K(backup_set_id),
+                      K(tablet_attr->tablet_meta_index_), K(tablet_attr->data_type_));
+        } else if ((OB_ISNULL(tablet_attr->sstable_meta_index_)
+                    && id_mappings_meta_.at(i)->sstable_count_ > 0)
+                   || (OB_NOT_NULL(tablet_attr->sstable_meta_index_)
+                       && 0 == id_mappings_meta_.at(i)->sstable_count_)) {
+          ret = OB_ERR_UNEXPECTED;
+          STORAGE_LOG(WARN, "empty tablet does not match non-empty tablet", K(ret));
+        } else if (OB_NOT_NULL(tablet_attr->sstable_meta_index_)
+                   && OB_FAIL(ObAdminBackupValidationUtil::read_sstable_metas(
+                       backup_set_attr->backup_set_store_->get_backup_set_dest(),
+                       *tablet_attr->sstable_meta_index_, tablet_attr->data_type_,
+                       inner_sstable_meta, ctx_))) {
+          STORAGE_LOG(WARN, "failed to read sstable meta", K(ret), K(backup_set_id),
+                      K(tablet_attr->tablet_meta_index_), K(tablet_attr->data_type_));
+        } else if (OB_FAIL(sstable_metas_.push_back(inner_sstable_meta))) {
+          STORAGE_LOG(WARN, "failed to push back sstable meta", K(ret), K(backup_set_id),
+                      K(tablet_attr->tablet_meta_index_), K(tablet_attr->data_type_));
+        } else {
+          STORAGE_LOG(INFO, "succeed to read sstable meta info", K(ret), K(backup_set_id));
+        }
+      }
+    }
+  }
+  if (OB_NOT_NULL(ctx_) && OB_FAIL(ret)) {
+    ctx_->go_abort("SSTable meta and Macro Block ID mappings meta info",
+                   "meta info is not complete");
+  }
+  return ret;
+}
+int ObAdminTabletMetaValidationTask::check_sstable_meta_info_and_prepare_macro_block_()
```

Comment:
I cannnot not understand this function...

Reply:
basically is to prepare macro block check. so cross-check meta info between sstable_meta and id_mapping_meta